playbook/app/pb_kits/playbook/pb_select/_select.jsx

```diff
@@ -126,17 +126,17 @@ const Select = ({
             {optionsList}
           </select>
         </If>
+        <Icon
```

Comment:
After all the debugging, what solves this is moving `Icon` above the `If` tag. Since `Icon` uses FontAwesome, a DOM bug was being introduced by their JS.